frontend/packages/console-plugin-sdk/src/typings/dev-catalog.ts

```diff
@@ -1,17 +1,115 @@
 import { K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
-import { Extension } from './base';
+import { CodeRef, Extension } from './base';
 
 namespace ExtensionProperties {
   export interface DevCatalogModel {
     model: K8sKind;
     normalize: (data: K8sResourceKind[]) => K8sResourceKind[];
   }
+
+  export interface CatalogItemType {
+    /** Type for the catalog item. */
+    type: string;
+    /** Title fpr the catalog item. */
+    title: string;
+    /** Catalog ID for which this provider contributes to. Idea here is to be able to support multiple catalogs. */
+    catalog: string;
+    /** Description for the type specific catalog. */
+    catalogDescription?: string;
+    /** Description for the catalog item type. */
+    typeDescription?: string;
+    /** Custom filters specific to the catalog item  */
+    filters?: CodeRef<CatalogExtensionHook<CatalogFilter[]>>;
+    /** Custom groupings specific to the catalog item */
+    groupings?: CodeRef<CatalogExtensionHook<CatalogGrouping[]>>;
```

Comment:
why do these need the `options` param?

Reply:
Could be useful in fetching items in the current namespace for some filters. That's why kept it same for all.

Reply:
Made filters and groupings simple descriptor types instead of hooks.